### ISSUE TITLE
Bote: AIOOBE (Trac #1603)

### ISSUE THREAD

**Issue by str4d** [i2p/i2p.i2p-bote#54]
I sometimes get 0 size files and those trigger errors liek this one: 

6/12/15 4:11:39 PM ERROR [ExpiratnThrd] i2p.bote.folder.Folder : Can't create a FolderElement from file: /path/to/.i2p/i2pbote/dht_index_pkt/nW6uWINdi~5TqZKzkG~YhzLf8evLrF0IWkqHpR091Rg=.pkt (file size=0)
     java.lang.ArrayIndexOutOfBoundsException: 0
     at i2p.bote.packet.CommunicationPacket.isPrefixValid(CommunicationPacket.java:122)
     at i2p.bote.packet.I2PBotePacket.createPacket(I2PBotePacket.java:115)
     at i2p.bote.packet.I2PBotePacket.createPacket(I2PBotePacket.java:90)
     at i2p.bote.folder.PacketFolder.createFolderElement(PacketFolder.java:85)
     at i2p.bote.folder.PacketFolder.createFolderElement(PacketFolder.java:42)
     at i2p.bote.folder.Folder$3.findNextElement(Folder.java:170)
     at i2p.bote.folder.Folder$3.next(Folder.java:151)
     at i2p.bote.folder.PacketFolder$1.next(PacketFolder.java:106)
     at i2p.bote.folder.PacketFolder$1.next(PacketFolder.java:90)
     at i2p.bote.folder.IndexPacketFolder.deleteExpired(IndexPacketFolder.java:99)
     at i2p.bote.service.ExpirationThread.run(ExpirationThread.java:53)

Migrated from https://trac.i2p2.de/ticket/1603
```json
{
    "status": "closed", 
    "changetime": "2015-11-15T05:27:46", 
    "description": "I sometimes get 0 size files and those trigger errors liek this one: \n\n6/12/15 4:11:39 PM ERROR [ExpiratnThrd] i2p.bote.folder.Folder : Can't create a FolderElement from file: /path/to/.i2p/i2pbote/dht_index_pkt/nW6uWINdi~5TqZKzkG~YhzLf8evLrF0IWkqHpR091Rg=.pkt (file size=0)\n     java.lang.ArrayIndexOutOfBoundsException: 0\n     at i2p.bote.packet.CommunicationPacket.isPrefixValid(CommunicationPacket.java:122)\n     at i2p.bote.packet.I2PBotePacket.createPacket(I2PBotePacket.java:115)\n     at i2p.bote.packet.I2PBotePacket.createPacket(I2PBotePacket.java:90)\n     at i2p.bote.folder.PacketFolder.createFolderElement(PacketFolder.java:85)\n     at i2p.bote.folder.PacketFolder.createFolderElement(PacketFolder.java:42)\n     at i2p.bote.folder.Folder$3.findNextElement(Folder.java:170)\n     at i2p.bote.folder.Folder$3.next(Folder.java:151)\n     at i2p.bote.folder.PacketFolder$1.next(PacketFolder.java:106)\n     at i2p.bote.folder.PacketFolder$1.next(PacketFolder.java:90)\n     at i2p.bote.folder.IndexPacketFolder.deleteExpired(IndexPacketFolder.java:99)\n     at i2p.bote.service.ExpirationThread.run(ExpirationThread.java:53)\n", 
    "reporter": "user", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1447565266287728", 
    "component": "apps/plugins", 
    "summary": "Bote: AIOOBE", 
    "priority": "minor", 
    "keywords": "I2P-Bote", 
    "version": "0.9.20", 
    "parents": "", 
    "time": "2015-06-12T18:02:38", 
    "milestone": "0.9.23", 
    "owner": "str4d", 
    "type": "defect"
}
```

**Comment by str4d** [comment]
Trac update at 20150623T14:01:49: 
* zzz changed component from "other" to "apps/plugins"
* zzz changed owner from "" to "str4d"
* zzz changed status from "new" to "assigned"

**Comment by str4d** [comment]
Trac update at 20151106T21:17:48: kay commented:

> In order to prevent this IAOOBE and a possible NPE check for null and packet array length.
> 
> 
> ```text
> in/java/i2p/bote/packet/CommunicationPacket.java 
> #
> # old_revision [0d6505d2805dff9774d816a899b6463074057c15]
> #
> # patch "src/main/java/i2p/bote/packet/CommunicationPacket.java"
> #  from [bb37a2da9cb08fdc7656f3daf0d17e3e4c008c0f]
> #    to [53b933a1f8fb872af0e0d95929509f51ac9eb456]
> #
> ============================================================
> --- src/main/java/i2p/bote/packet/CommunicationPacket.java	bb37a2da9cb08fdc7656f3daf0d17e3e4c008c0f
> +++ src/main/java/i2p/bote/packet/CommunicationPacket.java	53b933a1f8fb872af0e0d95929509f51ac9eb456
> @@ -118,6 +118,8 @@ public abstract class CommunicationPacke
>       * @param packet
>       */
>      static boolean isPrefixValid(byte[] packet) {
> +        if(null == packet || packet.length < PACKET_PREFIX.length)
> +            return false;
>          for (int i=0; i<PACKET_PREFIX.length; i++)
>              if (packet[i] != PACKET_PREFIX[i])
>                  return false;
> ```
> 
>

**Comment by str4d** [comment]
Trac update at 20151106T21:28:08: kay changed keywords from "" to "I2P-Bote needs_review"

**Comment by str4d** [comment]
Trac update at 20151115T05:27:46: 
* str4d commented:

> Fixes in `6b51a1f56403aeddc800bccf982181cebb65beae`.


* str4d changed keywords from "I2P-Bote needs_review" to "I2P-Bote"
* str4d changed milestone from "undecided" to "0.9.23"
* str4d changed resolution from "" to "fixed"
* str4d changed status from "assigned" to "closed"
